taggit/managers.py

```diff
@@ -666,14 +667,26 @@ def get_joining_columns(self, reverse_join=False):
         else:
             return (("object_id", self.model._meta.pk.column),)
 
-    def get_extra_restriction(self, where_class, alias, related_alias):
+    def _get_extra_restriction(self, alias, related_alias):
         extra_col = self.through._meta.get_field("content_type").column
         content_type_ids = [
             ContentType.objects.get_for_model(subclass).pk
             for subclass in _get_subclasses(self.model)
         ]
         return ExtraJoinRestriction(related_alias, extra_col, content_type_ids)
 
+    def _get_extra_restriction_legacy(self, where_class, alias, related_alias):
+        # this is a shim to maintain compatibility with django < 4.0
+        return self._get_extra_restriction(alias, related_alias)
+
+    # this is required to handle a change in Django 4.0
+    # https://docs.djangoproject.com/en/4.0/releases/4.0/#miscellaneous
+    # the signature of the (private) funtion was changed
+    if django.VERSION > (3, 8):
```

Comment:
3, 8 or 3, 2?

Reply:
Ah, woops, should be `(3, 2)`.

Reply:
Fixed this, I flipped it round to do ` < (4, 0)` instead.